File: 4_december/treeornament.js
function createXmasTree(height, ornament) {
    const createRow = (spaces, content) => '_'.repeat(spaces) + content + '_'.repeat(spaces);

    // Construir las partes del árbol
    const tree = Array.from({ length: height }, (_, i) => 
      createRow(height - i - 1, ornament.repeat(2 * i + 1))
    ).join('\n');

    const trunk = createRow(height - 1, '#');

    // Combinar el árbol con el tronco
    return `${tree}\n${trunk}\n${trunk}`;
}

console.log(createXmasTree(5, '·'))

File: 4_december/treeornament.py
def create_xmas_tree(height, ornament):
  create_row = lambda spaces, content: '_'*(spaces) + content + '_'*(spaces)
  create_tree = lambda n, content: content * n
  trunk = create_row(height-1, '#')
  tree = ''
  row = 1
  for i in range(height):
    tree += create_row(height-i-1, create_tree(row, ornament)) + '\n'
    row += 2
  return tree+trunk+'\n'+trunk

print(create_xmas_tree(3, '·'))